xarray/core/groupby.py

```diff
@@ -382,7 +382,15 @@ def fillna(self, value):
         Dataset.fillna
         DataArray.fillna
         """
-        return self._fillna(value)
+        def _yield_applied(this, other):
```

Comment:
One solution is perhaps to add something into `apply_groupby_ufunc` to make sure the `attrs` are copied over, similar to what's in `_copy_attrs_from` in dataset.py/dataarray.py.